server/datastore/mysql/operating_systems.go

```diff
@@ -136,26 +145,38 @@ func getOperatingSystemDB(ctx context.Context, tx sqlx.ExtContext, hostOS fleet.
 	return &os, nil
 }
 
-// upsertHostOperatingSystemDB upserts the host operating system table
-// with the operating system id for the given host ID
-func upsertHostOperatingSystemDB(ctx context.Context, tx sqlx.ExtContext, hostID uint, osID uint) error {
-	res, err := tx.ExecContext(ctx, "UPDATE host_operating_system SET os_id = ? WHERE host_id = ?", osID, hostID)
+func isHostOperatingSystemUpdateNeeded(ctx context.Context, qc sqlx.QueryerContext, hostID uint, hostOS fleet.OperatingSystem) (
+	bool, error,
+) {
+	rows, err := qc.QueryContext(
```

Comment:
Can we use `GetContext` or `SelectContext` here (if only for readability of the row/error handling)?

Reply:
Ok. makes sense.